includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -2063,6 +2066,8 @@ private function finalize_stylesheet_set( $stylesheet_set ) {
 									empty( $parsed_selector['tags'] )
 									||
 									0 === count( array_diff( $parsed_selector['tags'], $this->get_used_tag_names() ) )
+									||
+									0 === count( array_diff( $parsed_selector['tags'], $stylesheet_set['cdata_spec']['css_spec']['allowed_at_rules'] ) )
```

Comment:
This seemed to handle an issue with the Essence Pro theme, where the Ionicons stylesheet (snippet above) is removed entirely. But this might not be the best approach.